### DISCUSSION TITLE
Recent performance regression?

### DISCUSSION THREAD

**Discussion by wberrier** [gsantner/markor#1794] [General] [answered]
Within the last google play store update, I noticed a performance regression:

I have a 75k markdown file that is very slow when entering text now.

Any thoughts or ideas?

**Comment by foxandcity** [comment]
I have the same problem. Use Markor to edit fairly large plain text files (sometimes 200kb+) and while previous versions had no issue, newest release is unusable. Typing leads to one character inserted per second or so.

**Comment by harshad1** [comment]
200k files with highlighting have always been slow. The more highlighting the slower they will be. The issue is fundamental to how android's TextView works.

Please turn off highlighting and confirm that the performance is better.

My most recent changes add dynamic highlighting support. With them, I am able to edit 500k files just fine.

**Comment by harshad1** [comment]
Actually, on investigating this further, there is definitely something strange going on.

Test files which were really fast to edit a few days ago are now slow. Disabling highlighting and auto format does not fix the issue.

Further, scrolling is nice and fast until I make an edit, and then the scrolling becomes slow. Not sure what could be causing these issues. A quick look at profiling data shows nothing.

**Comment by harshad1** [comment]
@gsantner The performance regression is not present in 11937612c6aae98cf33171bbe7a624483fc54f45, right before master / AndroidX was merged back into partial highlighitng, and is present in bd1f3c8bc24114c7d7ddfc7af9ad4ee47601243d, the merge commit.

Looking at the diff to see what the cause may be. Hopefully its not just that AndroidX AppCompatEditText is slow...

**Reply by harshad1** [reply]
Unfortunately I don't see anything obvious :(

https://github.com/harshad1/markor/compare/11937612c6aae98cf33171bbe7a624483fc54f45...bd1f3c8bc24114c7d7ddfc7af9ad4ee47601243d

**Reply by harshad1** [reply]
@gsantner I will definitely need help investigating this

**Reply by gsantner** [reply]
Last time I checked the androidx edittext component souece code it looked like very little difference compared to raw EditText.

Read your linked bug, there is some emoji feature to turn off at androidX edittext?! When its really just that and improves/fixes performance..why not 😄 

Though, I dont have a Android 12 device, here on A.10 its not present to me. Also the AndroidX stuff was merged to your branch for a week or so as well right (and you didnt notice it either?)

**Reply by harshad1** [reply]
Didn't notice it earlier. I thought all performance issues had been solved.

**Reply by gsantner** [reply]
No worries, and of course not blaming. Thanks for the PR!

Currently have day job to do, I can't program throughout the day mostly.

**Comment by harshad1** [comment, marked as answer]
This is the exact issue we're seeing:

https://issuetracker.google.com/issues/215391131


-> #1801
